src/fs.ts

```diff
@@ -3,7 +3,7 @@ import * as sysfs from 'fs';
 export interface FS {
     existsSync(path : string | Buffer) : boolean;
     readFile(filename : string, encoding : string, callback : (err : NodeJS.ErrnoException, data : string) => void) : void;
-    readFileSync(filename : string, encoding : string) : string;
+    readFileSync(filename : string, encoding?: string) : string| Buffer;
```

Comment:
This signature feels hard to work with at a type level because the result will need a type assertion to be safe.  I suggest splitting out a separate member for reading raw buffers e.g. `readFileToBufferSync(filename : string) : Buffer` and leaving the existing function as it was.

Reply:
I advise to use a dedicate fs extra library which is compatible with node.js fs lib, see  https://github.com/jprichardson/node-fs-extra